internal/distro/fedora/distro.go

```diff
@@ -0,0 +1,913 @@
+
+	rd := distroMap[distroName]
+
+	// Architecture definitions
+	x86_64 := architecture{
+		name:     distro.X86_64ArchName,
+		distro:   &rd,
+		legacy:   "i386-pc",
+		bootType: distro.LegacyBootType,
+	}
+
+	aarch64 := architecture{
+		name:     distro.Aarch64ArchName,
+		distro:   &rd,
+		bootType: distro.UEFIBootType,
+	}
+
+	// Shared Services
+	iotServices := []string{
+		"NetworkManager.service",
+		"firewalld.service",
+		"rngd.service",
+		"sshd.service",
+		"zezere_ignition.timer",
+		"zezere_ignition_banner.service",
+		"greenboot-grub2-set-counter",
+		"greenboot-grub2-set-success",
+		"greenboot-healthcheck",
+		"greenboot-rpm-ostree-grub2-check-fallback",
+		"greenboot-status",
+		"greenboot-task-runner",
+		"redboot-auto-reboot",
+		"redboot-task-runner",
+		"parsec",
+		"dbus-parsec",
+	}
+
+	// Image Definitions
+	iotCommitImgType := imageType{
+		name:        "fedora-iot-commit",
+		nameAliases: []string{"iot-commit"},
+		filename:    "commit.tar",
+		mimeType:    "application/x-tar",
+		packageSets: map[string]packageSetFunc{
+			buildPkgsKey: iotBuildPackageSet,
+			osPkgsKey:    iotCommitPackageSet,
+		},
+		defaultImageConfig: &distro.ImageConfig{
+			EnabledServices: iotServices,
+		},
+		rpmOstree:        true,
+		pipelines:        edgeCommitPipelines,
+		buildPipelines:   []string{"build"},
+		payloadPipelines: []string{"ostree-tree", "ostree-commit", "commit-archive"},
+		exports:          []string{"commit-archive"},
+	}
+
+	iotOCIImgType := imageType{
+		name:        "fedora-iot-container",
+		nameAliases: []string{"iot-container"},
+		filename:    "container.tar",
+		mimeType:    "application/x-tar",
+		packageSets: map[string]packageSetFunc{
+			buildPkgsKey: iotBuildPackageSet,
+			osPkgsKey:    iotCommitPackageSet,
+			containerPkgsKey: func(t *imageType) rpmmd.PackageSet {
+				return rpmmd.PackageSet{
+					Include: []string{"nginx"},
+				}
+			},
+		},
+		defaultImageConfig: &distro.ImageConfig{
+			EnabledServices: iotServices,
+		},
+		rpmOstree:        true,
+		bootISO:          false,
+		pipelines:        iotContainerPipelines,
+		buildPipelines:   []string{"build"},
+		payloadPipelines: []string{"ostree-tree", "ostree-commit", "container-tree", "container"},
+		exports:          []string{"container"},
+	}
+
+	iotInstallerImgType := imageType{
+		name:        "fedora-iot-installer",
+		nameAliases: []string{"iot-installer"},
+		filename:    "installer.iso",
+		mimeType:    "application/x-iso9660-image",
+		packageSets: map[string]packageSetFunc{
+			// TODO: non-arch-specific package set handling for installers
+			// This image type requires build packages for installers and
+			// ostree/edge.  For now we only have x86-64 installer build
+			// package sets defined.  When we add installer build package sets
+			// for other architectures, this will need to be moved to the
+			// architecture and the merging will happen in the PackageSets()
+			// method like the other sets.
```

Comment:
This TODO is obsolete.  I realise it was copied from the `rhel90` package, but it's obsolete there as well :smile: 
(I should clean that up at some point)

Reply:
I'll remove it. Thanks!

Reply:
Done, removed